source/constructs/api/data_source/service.py

```diff
@@ -361,13 +381,338 @@ def sync_jdbc_connection(
+        raise BizException(MessageEnum.SOURCE_SUBNET_NOT_CONTAIN_NAT.get_code(),
+                           MessageEnum.SOURCE_SUBNET_NOT_CONTAIN_NAT.get_msg())
+    if not ec2_client.describe_availability_zones(Filters=[{'Name': 'zone-name', 'Values': [jdbc.network_availability_zone]}])["AvailabilityZones"]:
+        raise BizException(MessageEnum.SOURCE_AVAILABILITY_ZONE_NOT_EXISTS.get_code(),
+                           MessageEnum.SOURCE_AVAILABILITY_ZONE_NOT_EXISTS.get_msg())
+
+def sync(jdbc: SourceJDBCConnection):
+    glue_connection_name = f"jdbc-{jdbc.account_provider}-{jdbc.instance}-connection"
+    glue_database_name = f"jdbc-{jdbc.account_provider}-{jdbc.instance}-database"
+    crawler_name = f"jdbc-{jdbc.account_provider}-{jdbc.instance}-crawler"
+    state = crud.get_jdbc_connection_glue_state(jdbc.account_provider, jdbc.account_id, jdbc.region, jdbc.instance)
+    crawler_role_arn = __gen_role_arn(account_id=jdbc.account_id,
+                                      region=jdbc.region,
+                                      role_name='GlueDetectionJobRole')
+    crud.set_jdbc_connection_glue_state(jdbc.account_provider, jdbc.account_id, jdbc.region, jdbc.instance, ConnectionState.PENDING.value)
+    try:
+        if jdbc.secret is not None:
+            credentials = gen_credentials(jdbc.account_id)
+            secretsmanager = boto3.client('secretsmanager',
+                                          aws_access_key_id=credentials['AccessKeyId'],
+                                          aws_secret_access_key=credentials['SecretAccessKey'],
+                                          aws_session_token=credentials['SessionToken'],
+                                          region_name=jdbc.region
+                                          )
+            """ :type : pyboto3.secretsmanager """
+            secret_value = secretsmanager.get_secret_value(
+                SecretId=jdbc.secret
+            )
+            secret_values = json.loads(secret_value['SecretString'])
+
+            payload = {
+                "engine": jdbc.engine,
+                "host": jdbc.address,
+                "port": jdbc.port,
+                "username": secret_values['username'],
+                "password": secret_values['password'],
+            }
+            schema_list = __list_rds_schema(jdbc.account_id,
+                                            jdbc.region,
+                                            credentials,
+                                            jdbc.instance,
+                                            payload,
+                                            [jdbc.network_sg_id],
+                                            jdbc.network_subnet_id)
+            # schema_list = list_jdbc_schema(jdbc)
+
+            logger.info("sync_jdbc_connection schema_list :")
+            logger.info(schema_list)
+            if len(schema_list) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)